### PR TITLE
METRON-1027: Errant Log Message When No GeoIP Hit on Valid, External IPv4 Address

### PR DESCRIPTION
**Pull request by nickwallen**
When a valid public IPv4 address does not match a record in the Maxmind GeoIP database, a log message indicates that an error occurred with the GeoLite2 DB.  

```
2017-07-05 12:50:34.266 o.a.m.e.a.g.GeoLiteDatabase WARN Metron GeoLite2 DB encountered an error
com.maxmind.geoip2.exception.AddressNotFoundException: The address X.X.X.X is not in the database.
at com.maxmind.geoip2.DatabaseReader.get(DatabaseReader.java:157) ~[stormjar.jar:?]
at com.maxmind.geoip2.DatabaseReader.city(DatabaseReader.java:202) ~[stormjar.jar:?]
at org.apache.metron.enrichment.adapters.geo.GeoLiteDatabase.get(GeoLiteDatabase.java:136) [stormjar.jar:?]
at org.apache.metron.enrichment.adapters.geo.GeoAdapter.enrich(GeoAdapter.java:47) [stormjar.jar:?]
at org.apache.metron.enrichment.adapters.geo.GeoAdapter.enrich(GeoAdapter.java:31) [stormjar.jar:?]
```

Really no error has occurred with the GeoIP database.  Not having a hit in the database is an expected condition and processing should continue without the errant log message.

### Testing

To test this change, setup a Metron environment with GeoIP enrichment enabled.  Ingest some telemetry containing an valid, external IPv4 address that is not contained within the GeoIP database.   You can use an IP like `203.0.113.1` as the range 203.0.113.0/24 is reserved for "TEST-NET-3" and should never be locatable.  Before this PR, you will see the noted error message in the Enrichment topology logs.  With this change, you will not see the error message in your logs.

## Pull Request Checklist

- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron 
- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### CODE REVIEW

**Comment by nickwallen** [comment]
@justinleet Another thing I noticed while in the code is that a lookup of a private IP address is expected to return an empty map.  Since we're using an Optional, wouldn't it make more sense for it to return `Optional.empty()`?  

The javadocs say that `Optional.empty()` indicates a lookup miss, not an empty map.  I could either make it return `Optional.empty()` or I could update the docs so that they match current behavior.  Let me know what you think.

Thanks

File: metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/adapters/geo/GeoAdapter.java [hunk 1]
```diff
@@ -46,7 +46,6 @@ public JSONObject enrich(CacheKey value) {
     JSONObject enriched = new JSONObject();
     Optional<HashMap<String, String>> result = GeoLiteDatabase.INSTANCE.get(value.coerceValue(String.class));
     if(!result.isPresent()) {
-      _LOG.error("GEO Enrichment failure: {}", value.coerceValue(String.class));
```
Comment (justinleet) [inline metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/adapters/geo/GeoAdapter.java]: Can you make it log.debug()?  I think it's still useful to be able to tell what happened, it just shouldn't be error.

**Comment by justinleet** [comment]
@nickwallen Off the top of my head, I think it could be Optional.empty(), but I'd like to see a debug log added so we can easily tell the difference between "IP not found" and "It wasn't even something we should have looked up".  Double check the logic on the function, but it should essentially be the same thing.

**Comment by justinleet** [comment]
Specifically, I'd say add it in 
```
    if (isIneligibleAddress(ip, addr)) {
      return Optional.of(new HashMap());
    }
```

**Comment by ottobackwards** [comment]
+1 by inspection.  I agree that the log should be moved in level and not dropped

File: metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/adapters/geo/GeoAdapter.java (hunk 1 above)
Comment (nickwallen) [inline metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/adapters/geo/GeoAdapter.java]: We already get logging in this case from GeoLiteDatabase:167.  I didn't think we needed another.

**Comment by justinleet** [comment]
+1 by inspection.  @nickwallen is right that the log already occurs in the GeoLiteDatabase, so I'm okay with dropping it.

**Comment by nickwallen** [comment]
Thanks for the review guys.  Just trying to understand why Travis is reporting a unit test failure.  Its like it is not running the latest code.  Bah.

**Comment by nickwallen** [comment]
Oh, nevermind.  Got it.  Sorry, Travis.  Thank you for your service.

**Comment by nickwallen** [comment]
Also, ran this up on Full Dev and did some quick validation on `GEO_GET` just to be sure.

```
[Stellar]>>> GEO_GET("192.168.0.1")
2017-07-11 15:11:13,401 INFO  [AeshProcess: 3] stellar.GeoEnrichmentFunctions: Initializing GeoEnrichmentFunctions
2017-07-11 15:11:13,790 WARN  [AeshProcess: 3] util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2017-07-11 15:11:14,636 WARN  [AeshProcess: 3] shortcircuit.DomainSocketFactory: The short-circuit local reads feature cannot be used because libhadoop cannot be loaded.
{}
[Stellar]>>> GEO_GET("75.118.136.186")
{country=US, dmaCode=535, city=Columbus, postalCode=43230, latitude=40.0345, location_point=40.0345,-82.8695, locID=4509177, longitude=-82.8695}
```

**Pull request closed without merging**